core/src/main/java/org/elasticsearch/http/netty/NettyHttpRequest.java

```diff
@@ -69,6 +69,14 @@ public NettyHttpRequest(org.jboss.netty.handler.codec.http.HttpRequest request,
     @Override
     public Method method() {
         HttpMethod httpMethod = request.getMethod();
+
+        if (request.headers() != null) {
+            String methodOverride = request.headers().get("X-HTTP-Method-Override");
+            if (methodOverride != null) {
+                httpMethod = HttpMethod.valueOf(methodOverride);
+            }
+        }
+
```

Comment:
in general it feels really bad that we have to support this in our production code to work around some http client limitations that doesn't allow to send delete (and get?) with body...  the question should be if this makes sense in our production code regardless I think.